controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -489,6 +468,38 @@ func (r *StorageConsumerReconciler) reconcileConsumerConfigMap(
 				return err
 			}
 		}
+
+		if controllerRef != nil && controllerRef.UID == r.storageConsumer.UID {
```

Comment:
This is basically an else to the 2 cases above. 
```suggestion
		else {
```

I would also suggest to reorder the cases
1. `if controllerRef == nil`
2 `else if controllerRef.UID == r.storageConsumer.UID`
3. `else` (here we know that controllerRef is not consumer UID)